File: today2/sum-root-to-leaf-numbers.ts
/**
 * https://leetcode.com/problems/sum-root-to-leaf-numbers/description/
 */
export { sumNumbers, TreeNode };

function sumNumbers(root: TreeNode | null): number {
  return -1;
}

class TreeNode {
  val: number;
  left: TreeNode | null;
  right: TreeNode | null;
  constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
    this.val = val === undefined ? 0 : val;
    this.left = left === undefined ? null : left;
    this.right = right === undefined ? null : right;
  }
}

File: returned/python/today10/sum_root_to_leaf_numbers.py
# https://leetcode.com/problems/sum-root-to-leaf-numbers/description/
from typing import Optional, List

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        total = 0
        def dfs(root: TreeNode, path: List[int]) -> None:
            path.append(root.val)
            if root.left == None and root.right == None:
                nonlocal total
                total += int("".join([str(x) for x in path]))
            if root.left:
                dfs(root.left, path)
            if root.right:
                dfs(root.right, path)
            path.pop()
        dfs(root, [])
        return total